ldm/lightning.py

```diff
@@ -69,7 +75,12 @@ def __init__(
         self.to(dtype=torch.float16)
 
         if use_deepspeed:
-            deepspeed_injection(self.model, fp16=fp16, enable_cuda_graph=enable_cuda_graph)
+            if not package_available("deepspeed"):
+                logger.warn("You provided use_deepspeed=True but Deepspeed isn't installed. Skipping...")
+            elif _detect_cuda() not in ["80"]:
+                logger.warn("You provided `use_deepspeed=True` but Deepspeed isn't supported on your architecture. Skipping...")
+            else:
+                deepspeed_injection(self.model, fp16=fp16, enable_cuda_graph=enable_cuda_graph)
```

Comment:
Not for this PR but have we benchmarked the image quality for fp16 vs fp32? It's not significant for diffusers but not sure about our implementation

Reply:
I didn't see significant differences, but let me generate some samples next to each other.

Reply:
Added some images on the Notion page.